src/library/scala/runtime/BoxesRunTime.java

```diff
@@ -73,10 +73,20 @@ private static int typeCode(Object a) {
         return java.lang.Float.valueOf(f);
     }
 
+    private static final Double DOUBLE_ZERO = java.lang.Double.valueOf(0.0);
+    private static final Double DOUBLE_NEG_ZERO = java.lang.Double.valueOf(-0.0);
+    private static final Double DOUBLE_UNIT = java.lang.Double.valueOf(1.0);
+    private static final Double DOUBLE_NEG_UNIT = java.lang.Double.valueOf(-1.0);
+
     public static java.lang.Double boxToDouble(double d) {
-        // System.out.println("box " + d);
-        // (new Throwable()).printStackTrace();
-        return java.lang.Double.valueOf(d);
+        // Double can detect negative zeros, but primitive == cannot
+        // also note that Double.NaN.equals(Double.NaN) is true, unlike primitive
+        Double value = java.lang.Double.valueOf(d);
```

Comment:
This will always create the box, even if you end up using the interned box. What's the benefit, then?